docs/rfcs/0001-designing-in-the-open.md

```diff
@@ -0,0 +1,139 @@
+This causes a few problems:
+
+* Community concerns are less likely to have an impact as core maintainers have
+  already agreed on a solution.
+* Initial design is gated to Grafana Labs employees.
+* Historical design docs with context and discussions become hard to find both
+  internally and even harder publicly.
+
+In the spirit of building a broader community, this document proposes inverting
+the design process to be public by default.
+
+## Goals
+
+* Outline a process to support public-first design documents
+* Lower the barrier to entry to becoming a maintainer that is not employed at
+  Grafana Labs
+* Encourage a community to contribute input to design documents
+* Encourage a community to contribute their own design documents
+
+## Non-Goals
+
+* Make everything public. If something can't be public for confidentiality
+  reasons, it can remain internal to the company.
+* Identify strict rules for when an RFC or issue is appropriate.
+
+## Terminology
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL
+NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED",  "MAY", and
+"OPTIONAL" in this document are to be interpreted as described in
+[RFC 2119](https://datatracker.ietf.org/doc/html/rfc2119).
+
+## Proposal
+
+Public propsals come in two forms:
+
+* RFCs (i.e., this document) for larger changes
+* Issues proposals for smaller changes
+
+This document does not prescribe when something is a "larger change" or a
+"smaller change." We don't have enough experience yet to be able to
+strongly define categories, but these guidelines can be used for a gut check:
+
+* Larger changes require proposals that require thought and explore multiple
+  options, OR
+* Larger changes have a lasting effect on the project as a whole, OR
+* Larger changes touch multiple systems throughout the project or dramatic
+  changes to a single system.
+* Smaller changes are the inverse of the above.
+
+### RFCs
+
+RFCs MUST be valid markdown composed of at least the following:
+
+1. A title.
+2. A bulletpoint list of metadata that MUST include original authorship date
+   and the author's name. Author name SHOULD include their GitHub username as a
+   parenthethical.
+3. A summary section describing the context for what is being proposed.
+4. A list of goals and non-goals for the proposal.
+5. One or more sections dedicated to the proposal.
+
+RFCs MUST be placed into the `docs/rfcs` folder of the Grafana Agent repository.
+
+RFCs SHOULD have a `Terminology` section following the goals and non goals that
+conform to [RFC 2119](https://datatracker.ietf.org/doc/html/rfc2119).
+
+Sections dedicated to the proposal SHOULD include alternatives considered.
+
+#### RFC IDs
+
+The filename of the RFC MUST be prefixed with the ID of the RFC. If an RFC does
+not yet have an ID, the ID MAY be omitted until one is assigned to it.
+
+#### RFC Mutability
+
+Once an RFC is accepted, all non-metadata information MUST remain immutable.
```

Comment:
Immutability makes sense for IETF, but since all of these will live in Git with its own version history, maybe we don't need to be as restrictive here?

Or, maybe, we're ok with errata fixes to an RFC, but the spirit/proposal of the RFC should not change?

Reply:
I think there's probably still some value in having immutability. It allows you to use an RFC ID alone for what is being implemented. Otherwise, you'd have to say "RFC-ID as of YYYY-MM" to make it clear which iteration of a document you're referring to.

Reply:
Imo we probably would want to avoid immutability where possible but if there are valid reasons for changing something, I don't think we should restrict it. It's likely not as big as an issue as it would be for the IETF as you mention, where they may be dealing with protocol specifications and multiple implementers.

Reply:
If you want to copy IETF, RFCs MUST NOT be mutable. They can have addendums, corrections, references, be superseeded by others. But they can not change.

Reply:
Imagine a variant of semver where you can sometimes maybe change what a version number points to.